### PR TITLE
Engine pr

### PR DESCRIPTION
**Pull request by BenjiPestell**
# Added line cutting (and other bits)

[Jira Ticket](<link_to_jira_ticket>)
- [x] Ready for review

## Checklist
- [x] I have completed a self review
- [ ] I have updated the [spreadsheet](https://docs.google.com/spreadsheets/d/1277nPOKtBFOk_kCo870_zA460-qlw_kGnYlLPwtnadE/edit#gid=0)
- [ ] I have updated the jira ticket
- [x] I have added the relevant labels to this PR
- [ ] I have updated documentation (if applicable)
- [ ] I have run the unit tests suite and they pass
- [ ] I have updated the [change log](https://docs.google.com/document/d/1t3s6lQuUcug1Fj8E_i2gBJNvXN0-funwBArG2hsia6k/edit)

## Dependencies for merge
- [ ] [#<pull_request_number>]

### Visual Test
- [ ] Not applicable
- [x] Own computer
- [ ] Console 7"
- [ ] Console 10"

### Function Test
- [ ] Not applicable
- [x] Own computer
- [ ] Console 7"
- [ ] Console 10"

### Unit Tests
- [ ] Not applicable
- [x] Completed

## Screenshots (if applicable)

### CODE REVIEW

File: src/asmcnc/apps/drywall_cutter_app/engine.py [hunk 1]
```diff
@@ -561,7 +616,7 @@ def engine_run(self):
             # Read in data
             gcode_lines = self.find_and_read_gcode_file(self.source_folder_path, self.config.active_config.shape_type, self.config.active_cutter.diameter)
             gcode_cut_depth, gcode_z_safe_distance = self.extract_cut_depth_and_z_safe_distance(gcode_lines)
-            x_size, y_size = self.read_in_custom_shape_dimentions(gcode_lines)
+            x_size, y_size, _, _, = self.read_in_custom_shape_dimensions(gcode_lines)
```
[the remark this answers is not in the document]
Comment (BenjiPestell) [reply src/asmcnc/apps/drywall_cutter_app/engine.py:619]: read_in_custom_shape_dimensions returns 4 values x_dim, y_dim, x_min, y_min

File: src/asmcnc/apps/drywall_cutter_app/engine.py [hunk 2]
```diff
@@ -354,6 +403,8 @@ def replace_cut_depth_and_z_safe_distance(self, gcode_lines, gcode_cut_depth, gc
 
     #Add datum to each x and y move command
     def apply_datum_offset(self, gcode_lines, x_adjustment, y_adjustment):
+        if x_adjustment == 0 and y_adjustment == 0:
+            return gcode_lines
         adjusted_lines = []
         for line in gcode_lines:
             if line.startswith("G1Z") or line.startswith("G1 Z"):
```
[the remark this answers is not in the document]
Comment (BenjiPestell) [reply src/asmcnc/apps/drywall_cutter_app/engine.py:410]: I don't think I understand how regex would be applicable here

File: src/asmcnc/apps/drywall_cutter_app/engine.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenjiPestell) [reply src/asmcnc/apps/drywall_cutter_app/engine.py:619]: Oh the comma. silly me

File: src/asmcnc/apps/drywall_cutter_app/engine.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Masc999) [reply src/asmcnc/apps/drywall_cutter_app/engine.py:410]: r"G1\s*Z" so there can be as many whitespaces. But nothing needed for now (especially as WE create the gcode...)

**Review by Masc999 — APPROVED** [review]

File: src/asmcnc/apps/drywall_cutter_app/engine.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (BenjiPestell) [reply src/asmcnc/apps/drywall_cutter_app/engine.py:410]: Ah I see, I was confused by the selected lines for this comment

**Pull request merged**